src/units.cpp

```diff
@@ -461,43 +460,70 @@ void updateUnitsMap(const UnitsPtr &units, UnitsMap &unitsMap, double exp = 1.0)
                 } else {
                     auto refUnits = model->units(ref);
                     if ((refUnits == nullptr) || refUnits->isImport()) {
-                        unitsMap.clear();
-                        break;
+                        return false;
+                    }
+                    if (!updateUnitsMap(refUnits, unitsMap, uExp * exp)) {
+                        return false;
                     }
-                    updateUnitsMap(refUnits, unitsMap, uExp * exp);
                 }
             }
         }
     }
+    return true;
 }
 
-UnitsMap createUnitsMap(const UnitsPtr &units)
+UnitsMap createUnitsMap(const UnitsPtr &units, bool &isValid)
 {
     UnitsMap unitsMap;
-    updateUnitsMap(units, unitsMap);
+    isValid = true;
+    if (!updateUnitsMap(units, unitsMap)) {
+        isValid = false;
+        return unitsMap;
+    }
 
     // Checking for exponents of zero in the map, which can be removed.
-    bool requireDimensionless = false;
     auto it = unitsMap.begin();
     while (it != unitsMap.end()) {
         if (it->second == 0.0) {
             it = unitsMap.erase(it);
-            requireDimensionless = true;
         } else if (it->first == "dimensionless") {
-            it->second = 0.0;
-            ++it;
+            it = unitsMap.erase(it);
         } else {
             ++it;
         }
     }
-    if (requireDimensionless) {
-        unitsMap.emplace(std::make_pair("dimensionless", 0.0));
-    }
-
     return unitsMap;
 }
 
-bool Units::equivalent(const UnitsPtr &units1, const UnitsPtr &units2)
+bool Units::requiresImports() const
+{
+    // Function to check child unit dependencies for imports.
+    std::string ref;
+    std::string prefix;
+    double exponent;
+    double multiplier;
+    std::string id;
```

Comment:
Declare them only if we have a parent (i.e. no need to declare variables if they are not going to be used).

Reply:
Done.